tests/test_training/test_opt.py

```diff
@@ -39,16 +39,16 @@
 
 from mrmustard.math import Math
 
+settings.SEED = 42
 math = Math()
 
 
 @given(n=st.integers(0, 3))
 def test_S2gate_coincidence_prob(n):
     """Testing the optimal probability of obtaining |n,n> from a two mode squeezed vacuum"""
-    tf.random.set_seed(137)
     S = S2gate(
-        r=abs(np.random.normal()),
-        phi=np.random.normal(),
+        r=abs(settings.rng.normal()),
```

Comment:
Should this test set the seed in the beggining like the other tests as well?

Reply:
there is a global seed set at the top of the file, but I moved the seeds into each test